contracts/core/interfaces/ICore.sol

```diff
@@ -268,55 +291,85 @@ interface ICore {
-     * @param  sigBytes                   Array with r and s segments of ECDSA signature
-     * @param _orderData                  Bytes array containing the exchange orders to execute
+     * @param  _from              Address to transfer tokens from
+     * @param  _to                Address to credit for deposits
+     * @param  _tokens            Array of the addresses of the tokens being deposited
+     * @param  _quantities        Array of the amounts of tokens to deposit
      */
-    function fillOrder(
-        address[5] _addresses,
-        uint[5] _values,
-        address[] _requiredComponents,
-        uint[] _requiredComponentAmounts,
-        uint _fillQuantity,
-        uint8 _v,
-        bytes32[] sigBytes,
-        bytes _orderData
+    function batchDepositModule(
+        address _from,
+        address _to,
+        address[] _tokens,
+        uint256[] _quantities
     )
         external;
 
     /**
-     * Cancel an issuance order
+     * Exposes internal function that withdraws multiple tokens from the vault, to system
+     * modules. Quantities should be in the order of the addresses of the tokens being withdrawn.
      *
-     * @param  _addresses                 [setAddress, makerAddress, makerToken, relayerAddress, relayerToken]
-     * @param  _values                    [quantity, makerTokenAmount, expiration, relayerTokenAmount, salt]
-     * @param  _requiredComponents        Components required for the issuance order
-     * @param  _requiredComponentAmounts  Component amounts required for the issuance order
-     * @param  _cancelQuantity            Quantity of set to be canceled
+     * @param  _from              Address to decredit for withdrawals
+     * @param  _to                Address to transfer tokens to
+     * @param  _tokens            Array of the addresses of the tokens being withdrawn
+     * @param  _quantities        Array of the amounts of tokens to withdraw
      */
-    function cancelOrder(
-        address[5] _addresses,
-        uint[5] _values,
-        address[] _requiredComponents,
-        uint[] _requiredComponentAmounts,
-        uint _cancelQuantity
+    function batchWithdrawModule(
+        address _from,
+        address _to,
+        address[] _tokens,
+        uint256[] _quantities
     )
         external;
 
     /**
-     * Bid on rebalancing a given quantity of sets held by a rebalancing token
+     * Expose internal function that exchanges components for Set tokens,
+     * accepting any owner, to system modules
      *
-     * @param  _rebalancingSetToken        The address of the rebalancing token being bid on
-     * @param  _quantity                   The number of currentSets to rebalance
+     * @param  _owner        Address to issue tokens to
+     * @param  _set          Address of the Set to issue
+     * @param  _quantity     Number of tokens to issue
      */
-    function bid(
-        address _rebalancingSetToken,
+    function issueModule(
```

Comment:
If we don't upgrade Core, we might want to expose other functions like `redeemInternal` as well

Reply:
+1